lib/dialects/oracle/connection-manager.js

```diff
@@ -0,0 +1,168 @@
+
+  constructor(dialect, sequelize) {
+    super(dialect, sequelize);
+
+    this.sequelize = sequelize;
+    this.sequelize.config.port = this.sequelize.config.port || 1521;
+    try {
+      if (sequelize.config.dialectModulePath) {
+        this.lib = require(sequelize.config.dialectModulePath);
+      } else {
+        this.lib = require('oracledb');
+        this.lib.maxRows = 1000;
+
+        if (sequelize.config && 'dialectOptions' in sequelize.config) {
+          const dialectOptions = sequelize.config.dialectOptions;
+          if (dialectOptions && 'maxRows' in dialectOptions) {
+            this.lib.maxRows = sequelize.config.dialectOptions.maxRows;
+          }
+
+          if (dialectOptions && 'fetchAsString' in dialectOptions) {
+            this.lib.fetchAsString = sequelize.config.dialectOptions.fetchAsString;
+          } else {
+            this.lib.fetchAsString = [this.lib.CLOB];
+          }
+        }
+        this.lib.Promise = Promise;
+      }
+    } catch (err) {
+      if (err.code === 'MODULE_NOT_FOUND') {
+        throw new Error('Please install oracledb package manually');
+      }
+      throw err;
+    }
+
+  }
+
+  /**
+  * Method for checking the config object passed and generate the full database if not fully passed
+  * With dbName, host and port, it generates a string like this : 'host:port/dbname' 
+  */
+  checkConfigObject(config) {
+    //A connectString should be defined
+    if (config.database.length === 0) {
+      let errorToThrow = 'The database cannot be blank, you must specify the database name (which correspond to the service name';
+      errorToThrow += '\n from tnsnames.ora : (HOST = mymachine.example.com)(PORT = 1521)(SERVICE_NAME = orcl)'; 
+      throw new Error(errorToThrow);
+    }
+
+    if (!config.host || config.host.length === 0) {
+      throw new Error('You have to specify the host');
+    }
+
+    //The connectString has a special format, we check it
+    //ConnectString format is : host:[port]/service_name
+    if (config.database.indexOf('/') === - 1) {
+      
+      let connectString = config.host;
+
+      if (config.port && config.port !== 0) {
+        connectString += `:${config.port}`;
+      } else {
+        connectString += ':1521'; //Default port number
+      }
+      connectString += `/${config.database}`;
+      config.database = connectString;
+    }
+  }
+
+  // Expose this as a method so that the parsing may be updated when the user has added additional, custom types
+  _refreshTypeParser(dataType) {
+    parserStore.refresh(dataType);
+  }
+
+  _clearTypeParser() {
+    parserStore.clear();
+  }
+
+  connect(config) {
+    const connectionConfig = {
+      user: config.username,
+      host:  config.host,
+      port : config.port,
+      database : config.database,
+      password: config.password,
+      externalAuth : config.externalAuth,
+      stmtCacheSize : 0
+    };
+
+    //We check if there dialect options
+    if ('dialectOptions' in config) {
```

Comment:
just use `config.dialectOptions`, same check exists about 10 lines below, good idea to cover both of them under same if check